backend/layers/persistence/persistence.py

```diff
@@ -507,6 +507,17 @@ def add_collection_version(self, collection_id: CollectionId) -> CollectionVersi
             session.add(new_version)
             return CollectionVersionId(new_version_id)
 
+    def delete_collection(self, collection_id: CollectionId) -> None:
```

Comment:
change to ```suggestion
    def delete_unpublished_collection(self, collection_id: CollectionId) -> None:
```
Otherwise don't mention the unpublished part since it add business logic to the persistence layer. Not even sure if you should be checking for published here since that is also business layer logic.

Reply:
Good points. I added the check for published/not published because I was following the precedent set by Emanuele's original code for `delete_collection_version` which is immediately below.

But yeah I will update the name.